src/paint/paint.c

```diff
@@ -429,9 +429,9 @@ void sub_685EBC(money32 amount, uint16 string_id, sint16 y, sint16 z, sint8 y_of
 	ps->string_id = string_id;
 	ps->next = 0;
 	ps->args[0] = amount;
-	ps->args[4] = y;
-	ps->args[8] = y_offsets;
-	ps->args[12] = 0;
+	ps->args[1] = y;
+	ps->args[2] = y_offsets;
```

Comment:
Can you add casts to supress the warnings. Ps. This is going to mess up on x64 I expect.